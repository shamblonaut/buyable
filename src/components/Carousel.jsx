import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";

const Container = styled.div`
  position: relative;
  width: 100%;
  overflow: hidden;
  padding: 16px;
  border: 2px solid black;
`;

const ItemList = styled.ul`
  list-style: none;
  padding: 0;
  display: flex;
  min-width: min-content;
  transform: translateX(${(props) => props.$activeIndex * -100}%);
  transition: all 500ms ease-in-out;
`;

const Item = styled.li`
  min-width: 100%;
  min-height: 160px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const CarouselDots = styled.div`
  list-style: none;
  padding: 0;
  display: flex;
  align-items: center;
  background-color: #00000080;
  height: 16px;
  padding: 0 4px;
  border-radius: 8px;

  position: absolute;
  bottom: 8px;
  left: 50%;
  transform: translateX(-50%);
`;

const CarouselDot = styled.button`
  appearance: none;
  border: none;
  cursor: pointer;
  padding: 0;
  background-color: ${(props) => (props.$active ? "#efefef" : "#efefef40")};
  width: ${(props) => (props.$active ? "8px" : "4px")};
  height: ${(props) => (props.$active ? "8px" : "4px")};
  border-radius: 50%;
  margin: 4px;

  &:hover {
    background-color: ${(props) => (props.$active ? "#efefef" : "#efefef80")};
    width: 8px;
    height: 8px;
  }
`;

const CarouselLink = styled(Link)`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const CarouselImage = styled.img`
  width: auto;
  height: auto;
  max-width: 80%;
  max-height: 250px;
`;

const Carousel = ({ items }) => {
  const [activeIndex, setActiveIndex] = useState(0);

  useEffect(() => {
    const timer = setTimeout(() => {
      if (activeIndex === items.length - 1) {
        setActiveIndex(0);
      } else {
        setActiveIndex(activeIndex + 1);
      }
    }, 5000);

    return () => clearTimeout(timer);
  }, [items, activeIndex]);

  return (
    <Container>
      <ItemList $activeIndex={activeIndex}>
        {items.map((item) => (
          <Item key={item.id}>
            <CarouselLink to={`/product/${item.id}`}>
              <CarouselImage src={item.image} alt={item.title} />
            </CarouselLink>
          </Item>
        ))}
      </ItemList>
      <CarouselDots>
        {items.map((item, index) => (
          <CarouselDot
            $active={activeIndex === index}
            key={item.id}
            onClick={() => setActiveIndex(index)}
          ></CarouselDot>
        ))}
      </CarouselDots>
    </Container>
  );
};

export default Carousel;
