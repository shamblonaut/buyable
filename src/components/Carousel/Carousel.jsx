import { useEffect, useState } from "react";
import PropTypes from "prop-types";

import {
  Info,
  Container,
  ItemList,
  Item,
  ContentNavigator,
  CarouselDot,
  ProductLink,
  ImageContainer,
  CarouselImage,
  ProductDetails,
  ProductTitle,
} from "./Carousel.styles";

const Carousel = ({ productsData, itemCount }) => {
  const products = productsData.data?.slice(
    0,
    itemCount
      ? Math.min(itemCount, productsData.data?.length)
      : productsData.data?.length,
  );

  const [activeIndex, setActiveIndex] = useState(0);

  useEffect(() => {
    if (!products) return;

    const timer = setTimeout(() => {
      setActiveIndex((previousIndex) =>
        previousIndex === products.length - 1 ? 0 : previousIndex + 1,
      );
    }, 5000);

    return () => clearTimeout(timer);
  });

  return (
    <Container>
      {productsData.loading ? (
        <Info>Loading products...</Info>
      ) : productsData.error ? (
        <Info>
          <p>Could not load carousel :(</p>
          <p>Error: {productsData.error.message}</p>
        </Info>
      ) : (
        <>
          <ItemList $activeIndex={activeIndex} aria-label="carousel-image-list">
            {products.map((product) => (
              <Item key={product.id}>
                <ProductLink
                  to={`/product/${product.id}`}
                  aria-label="product-link"
                >
                  <ImageContainer>
                    <CarouselImage src={product.image} alt={product.title} />
                  </ImageContainer>
                  <ProductDetails>
                    <ProductTitle>{product.title}</ProductTitle>
                  </ProductDetails>
                </ProductLink>
              </Item>
            ))}
          </ItemList>
          <ContentNavigator>
            {products.map((item, index) => (
              <CarouselDot
                $active={activeIndex === index}
                key={item.id}
                aria-label="carousel-navigation-button"
                onClick={() => setActiveIndex(index)}
              ></CarouselDot>
            ))}
          </ContentNavigator>
        </>
      )}
    </Container>
  );
};

Carousel.propTypes = {
  productsData: PropTypes.exact({
    data: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        title: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired,
      }),
    ),
    error: PropTypes.instanceOf(Error),
    loading: PropTypes.bool.isRequired,
  }).isRequired,
  itemCount: PropTypes.number,
};

export default Carousel;
