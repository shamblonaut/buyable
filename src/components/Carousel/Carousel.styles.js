import { Link } from "react-router-dom";
import styled from "styled-components";

import { Info as CommonInfo } from "@/styles";

export const Info = styled(CommonInfo)`
  flex: 1;
`;

export const Container = styled.div`
  position: relative;
  width: min(500px, 80%);
  min-height: 200px;
  overflow: hidden;
  padding: 16px;
  margin: 32px 0;

  background-color: white;
  border: 3px solid var(--color-dark);
  border-radius: 8px;

  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const ItemList = styled.ul`
  padding: 0;
  display: flex;
  list-style: none;
  min-width: min-content;
  transform: translateX(${(props) => props.$activeIndex * -100}%);
  transition: all 500ms ease-in-out;
`;

export const Item = styled.li`
  width: 100%;
  min-height: 160px;
  flex-shrink: 0;

  display: flex;
  justify-content: center;
  align-items: center;
`;

export const ContentNavigator = styled.div`
  list-style: none;
  padding: 0;

  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: 16px;
  align-items: center;
  justify-items: center;

  background-color: #00000080;
  height: 16px;
  padding: 0 4px;
  border-radius: 8px;

  position: absolute;
  bottom: 8px;
  left: 50%;
  transform: translateX(-50%);
`;

export const CarouselDot = styled.button`
  border: none;
  background: none;
  padding: 0;

  width: 100%;
  height: 100%;
  position: relative;

  &::after {
    position: absolute;
    content: "";
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: ${(props) => (props.$active ? "8px" : "4px")};
    height: ${(props) => (props.$active ? "8px" : "4px")};
    background-color: ${(props) =>
      props.$active ? "var(--color-light)" : "var(--color-light-dimmer)"};
    border-radius: 50%;
  }

  &:hover::after {
    background-color: ${(props) =>
      props.$active ? "var(--color-light)" : "var(--color-light-dim)"};

    width: 8px;
    height: 8px;
  }
`;

export const ProductLink = styled(Link)`
  width: 100%;
  height: 100%;

  display: flex;
  justify-content: center;
  align-items: center;
`;

export const CarouselImage = styled.img`
  width: auto;
  height: auto;
  max-width: 80%;
  max-height: 250px;
`;
