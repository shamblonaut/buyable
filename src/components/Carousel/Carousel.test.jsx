import { describe, it, expect, vi, beforeEach, afterEach } from "vitest";
import { act, render, screen, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { BrowserRouter } from "react-router-dom";

import Carousel from "./Carousel";

const mockProductList = [
  { id: 1, title: "Product 1", image: "http://example.com/image-1.jpg" },
  { id: 2, title: "Product 2", image: "http://example.com/image-2.jpg" },
  { id: 3, title: "Product 3", image: "http://example.com/image-3.jpg" },
  { id: 4, title: "Product 4", image: "http://example.com/image-4.jpg" },
  { id: 5, title: "Product 5", image: "http://example.com/image-5.jpg" },
  { id: 6, title: "Product 6", image: "http://example.com/image-6.jpg" },
  { id: 7, title: "Product 7", image: "http://example.com/image-7.jpg" },
  { id: 8, title: "Product 8", image: "http://example.com/image-8.jpg" },
  { id: 9, title: "Product 9", image: "http://example.com/image-9.jpg" },
  { id: 10, title: "Product 10", image: "http://example.com/image-10.jpg" },
];

describe("Carousel component", () => {
  beforeEach(() => {
    vi.useFakeTimers();
  });

  afterEach(() => {
    vi.runOnlyPendingTimers();
    vi.useRealTimers();
  });

  it("renders correctly when products data is available", () => {
    const mockData = {
      data: mockProductList,
      error: null,
      loading: false,
    };

    const { container } = render(
      <BrowserRouter>
        <Carousel productsData={mockData} />
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  it("shows loading status", () => {
    const mockData = {
      data: null,
      error: null,
      loading: true,
    };

    const { container } = render(
      <BrowserRouter>
        <Carousel productsData={mockData} />
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
    expect(
      screen.getByText("loading products", { exact: false }),
    ).toBeInTheDocument();
  });

  it("shows error status when products data is not available", () => {
    const mockData = {
      data: null,
      error: new Error("Could not fetch products"),
      loading: false,
    };

    const { container } = render(
      <BrowserRouter>
        <Carousel productsData={mockData} />
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
    expect(
      screen.getByText("Could not load carousel", { exact: false }),
    ).toBeInTheDocument();
    expect(screen.getByText("error", { exact: false })).toBeInTheDocument();
  });

  it("displays a list of products", () => {
    const mockData = {
      data: mockProductList,
      error: null,
      loading: false,
    };

    render(
      <BrowserRouter>
        <Carousel productsData={mockData} />
      </BrowserRouter>,
    );

    expect(screen.getAllByRole("listitem").length).toBe(mockData.data.length);
  });

  it("displays only given no. of items if mentioned", () => {
    const mockData = {
      data: mockProductList,
      error: null,
      loading: false,
    };
    const itemCount = 4;

    render(
      <BrowserRouter>
        <Carousel productsData={mockData} itemCount={itemCount} />
      </BrowserRouter>,
    );

    expect(screen.getAllByRole("listitem").length).toBe(itemCount);
  });

  it("works when item count is larger than no. of products", () => {
    const mockData = {
      data: mockProductList,
      error: null,
      loading: false,
    };
    const itemCount = 16;

    render(
      <BrowserRouter>
        <Carousel productsData={mockData} itemCount={itemCount} />
      </BrowserRouter>,
    );

    expect(screen.getAllByRole("listitem").length).toBe(mockData.data.length);
  });

  it("links each image to its product page", () => {
    const mockData = {
      data: mockProductList,
      error: null,
      loading: false,
    };

    render(
      <BrowserRouter>
        <Carousel productsData={mockData} />
      </BrowserRouter>,
    );

    expect(
      within(screen.getAllByRole("listitem")[0]).getByRole("link"),
    ).toHaveAttribute("href", `/product/${mockData.data[0].id}`);
  });

  it("has navigation buttons corresponding to each product", () => {
    const mockData = {
      data: mockProductList,
      error: null,
      loading: false,
    };

    render(
      <BrowserRouter>
        <Carousel productsData={mockData} />
      </BrowserRouter>,
    );

    expect(
      screen.getAllByRole("button", { name: "carousel-navigation-button" })
        .length,
    ).toBe(mockData.data.length);
  });

  it("shows corresponding product when navigation button is clicked", async () => {
    const user = userEvent.setup({ advanceTimers: vi.advanceTimersByTime });
    const mockData = {
      data: mockProductList,
      error: null,
      loading: false,
    };

    render(
      <BrowserRouter>
        <Carousel productsData={mockData} />
      </BrowserRouter>,
    );

    expect(
      screen.getByRole("list", { name: "carousel-image-list" }),
    ).toHaveStyle({
      transform: `translateX(0%)`,
    });

    await act(async () => {
      user.click(
        screen.getAllByRole("button", {
          name: "carousel-navigation-button",
        })[3],
      );
    });

    expect(
      screen.getByRole("list", { name: "carousel-image-list" }),
    ).toHaveStyle({
      transform: `translateX(-300%)`,
    });
  });

  it("automatically scrolls products after a 5s delay", () => {
    const mockData = {
      data: mockProductList,
      error: null,
      loading: false,
    };

    render(
      <BrowserRouter>
        <Carousel productsData={mockData} />
      </BrowserRouter>,
    );

    expect(
      screen.getByRole("list", { name: "carousel-image-list" }),
    ).toHaveStyle({
      transform: `translateX(0%)`,
    });

    act(() => {
      vi.advanceTimersByTime(5000);
    });

    expect(
      screen.getByRole("list", { name: "carousel-image-list" }),
    ).toHaveStyle({
      transform: `translateX(-100%)`,
    });

    vi.clearAllTimers();
  });

  it("goes back to first item after the final one", async () => {
    const mockData = {
      data: mockProductList,
      error: null,
      loading: false,
    };

    render(
      <BrowserRouter>
        <Carousel productsData={mockData} />
      </BrowserRouter>,
    );

    expect(
      screen.getByRole("list", { name: "carousel-image-list" }),
    ).toHaveStyle({ transform: `translateX(0%)` });

    for (let i = 1; i < mockProductList.length; i++) {
      act(() => {
        vi.advanceTimersByTime(5000 * i);
      });

      expect(
        screen.getByRole("list", { name: "carousel-image-list" }),
      ).toHaveStyle({
        transform: `translateX(-${100 * i}%)`,
      });
    }

    act(() => {
      vi.advanceTimersByTime(5000);
    });

    expect(
      screen.getByRole("list", { name: "carousel-image-list" }),
    ).toHaveStyle({
      transform: `translateX(0%)`,
    });

    vi.clearAllTimers();
  });

  it("waits another 5s after click navigation before auto-scroll", async () => {
    const user = userEvent.setup({ advanceTimers: vi.advanceTimersByTime });
    const mockData = {
      data: mockProductList,
      error: null,
      loading: false,
    };

    render(
      <BrowserRouter>
        <Carousel productsData={mockData} />
      </BrowserRouter>,
    );

    expect(
      screen.getByRole("list", { name: "carousel-image-list" }),
    ).toHaveStyle({
      transform: `translateX(0%)`,
    });

    act(() => {
      vi.advanceTimersByTime(3000);
    });

    await act(async () => {
      user.click(
        screen.getAllByRole("button", {
          name: "carousel-navigation-button",
        })[3],
      );
    });

    expect(
      screen.getByRole("list", { name: "carousel-image-list" }),
    ).toHaveStyle({
      transform: `translateX(-300%)`,
    });

    act(() => {
      vi.advanceTimersByTime(2000);
    });

    expect(
      screen.getByRole("list", { name: "carousel-image-list" }),
    ).toHaveStyle({
      transform: `translateX(-300%)`,
    });

    act(() => {
      vi.advanceTimersByTime(5000);
    });

    expect(
      screen.getByRole("list", { name: "carousel-image-list" }),
    ).toHaveStyle({
      transform: `translateX(-400%)`,
    });

    vi.clearAllTimers();
  });
});
