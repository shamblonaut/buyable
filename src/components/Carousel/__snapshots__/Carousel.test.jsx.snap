// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carousel component > renders correctly when products data is available 1`] = `
<div>
  <div
    class="sc-dTvVRJ bRYsGP"
  >
    <ul
      aria-label="carousel-image-list"
      class="sc-jwTyAe iHTzTy"
    >
      <li
        class="sc-hjsuWn ibuNrn"
      >
        <a
          aria-label="product-link"
          class="sc-jJLAfE fulvdC"
          data-discover="true"
          href="/product/1"
        >
          <div
            class="sc-hwkwBN dYxlyM"
          >
            <img
              alt="Product 1"
              class="sc-kNOymR VXMnK"
              src="http://example.com/image-1.jpg"
            />
          </div>
          <div
            class="sc-dYwGCk fQsBUW"
          >
            <p
              class="sc-lgpSej eNDOWy"
            >
              Product 1
            </p>
          </div>
        </a>
      </li>
      <li
        class="sc-hjsuWn ibuNrn"
      >
        <a
          aria-label="product-link"
          class="sc-jJLAfE fulvdC"
          data-discover="true"
          href="/product/2"
        >
          <div
            class="sc-hwkwBN dYxlyM"
          >
            <img
              alt="Product 2"
              class="sc-kNOymR VXMnK"
              src="http://example.com/image-2.jpg"
            />
          </div>
          <div
            class="sc-dYwGCk fQsBUW"
          >
            <p
              class="sc-lgpSej eNDOWy"
            >
              Product 2
            </p>
          </div>
        </a>
      </li>
      <li
        class="sc-hjsuWn ibuNrn"
      >
        <a
          aria-label="product-link"
          class="sc-jJLAfE fulvdC"
          data-discover="true"
          href="/product/3"
        >
          <div
            class="sc-hwkwBN dYxlyM"
          >
            <img
              alt="Product 3"
              class="sc-kNOymR VXMnK"
              src="http://example.com/image-3.jpg"
            />
          </div>
          <div
            class="sc-dYwGCk fQsBUW"
          >
            <p
              class="sc-lgpSej eNDOWy"
            >
              Product 3
            </p>
          </div>
        </a>
      </li>
      <li
        class="sc-hjsuWn ibuNrn"
      >
        <a
          aria-label="product-link"
          class="sc-jJLAfE fulvdC"
          data-discover="true"
          href="/product/4"
        >
          <div
            class="sc-hwkwBN dYxlyM"
          >
            <img
              alt="Product 4"
              class="sc-kNOymR VXMnK"
              src="http://example.com/image-4.jpg"
            />
          </div>
          <div
            class="sc-dYwGCk fQsBUW"
          >
            <p
              class="sc-lgpSej eNDOWy"
            >
              Product 4
            </p>
          </div>
        </a>
      </li>
      <li
        class="sc-hjsuWn ibuNrn"
      >
        <a
          aria-label="product-link"
          class="sc-jJLAfE fulvdC"
          data-discover="true"
          href="/product/5"
        >
          <div
            class="sc-hwkwBN dYxlyM"
          >
            <img
              alt="Product 5"
              class="sc-kNOymR VXMnK"
              src="http://example.com/image-5.jpg"
            />
          </div>
          <div
            class="sc-dYwGCk fQsBUW"
          >
            <p
              class="sc-lgpSej eNDOWy"
            >
              Product 5
            </p>
          </div>
        </a>
      </li>
      <li
        class="sc-hjsuWn ibuNrn"
      >
        <a
          aria-label="product-link"
          class="sc-jJLAfE fulvdC"
          data-discover="true"
          href="/product/6"
        >
          <div
            class="sc-hwkwBN dYxlyM"
          >
            <img
              alt="Product 6"
              class="sc-kNOymR VXMnK"
              src="http://example.com/image-6.jpg"
            />
          </div>
          <div
            class="sc-dYwGCk fQsBUW"
          >
            <p
              class="sc-lgpSej eNDOWy"
            >
              Product 6
            </p>
          </div>
        </a>
      </li>
      <li
        class="sc-hjsuWn ibuNrn"
      >
        <a
          aria-label="product-link"
          class="sc-jJLAfE fulvdC"
          data-discover="true"
          href="/product/7"
        >
          <div
            class="sc-hwkwBN dYxlyM"
          >
            <img
              alt="Product 7"
              class="sc-kNOymR VXMnK"
              src="http://example.com/image-7.jpg"
            />
          </div>
          <div
            class="sc-dYwGCk fQsBUW"
          >
            <p
              class="sc-lgpSej eNDOWy"
            >
              Product 7
            </p>
          </div>
        </a>
      </li>
      <li
        class="sc-hjsuWn ibuNrn"
      >
        <a
          aria-label="product-link"
          class="sc-jJLAfE fulvdC"
          data-discover="true"
          href="/product/8"
        >
          <div
            class="sc-hwkwBN dYxlyM"
          >
            <img
              alt="Product 8"
              class="sc-kNOymR VXMnK"
              src="http://example.com/image-8.jpg"
            />
          </div>
          <div
            class="sc-dYwGCk fQsBUW"
          >
            <p
              class="sc-lgpSej eNDOWy"
            >
              Product 8
            </p>
          </div>
        </a>
      </li>
      <li
        class="sc-hjsuWn ibuNrn"
      >
        <a
          aria-label="product-link"
          class="sc-jJLAfE fulvdC"
          data-discover="true"
          href="/product/9"
        >
          <div
            class="sc-hwkwBN dYxlyM"
          >
            <img
              alt="Product 9"
              class="sc-kNOymR VXMnK"
              src="http://example.com/image-9.jpg"
            />
          </div>
          <div
            class="sc-dYwGCk fQsBUW"
          >
            <p
              class="sc-lgpSej eNDOWy"
            >
              Product 9
            </p>
          </div>
        </a>
      </li>
      <li
        class="sc-hjsuWn ibuNrn"
      >
        <a
          aria-label="product-link"
          class="sc-jJLAfE fulvdC"
          data-discover="true"
          href="/product/10"
        >
          <div
            class="sc-hwkwBN dYxlyM"
          >
            <img
              alt="Product 10"
              class="sc-kNOymR VXMnK"
              src="http://example.com/image-10.jpg"
            />
          </div>
          <div
            class="sc-dYwGCk fQsBUW"
          >
            <p
              class="sc-lgpSej eNDOWy"
            >
              Product 10
            </p>
          </div>
        </a>
      </li>
    </ul>
    <div
      class="sc-eqYatC jVAfUX"
    >
      <button
        aria-label="previous-slide-button"
        class="sc-kcLKEh grrYbN"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-chevron-left"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m15 18-6-6 6-6"
          />
        </svg>
      </button>
      <div
        class="sc-dntSTA dGzGdu"
      >
        <button
          aria-label="carousel-navigation-button"
          class="sc-kvnevz clbvBz"
        />
        <button
          aria-label="carousel-navigation-button"
          class="sc-kvnevz kinrjB"
        />
        <button
          aria-label="carousel-navigation-button"
          class="sc-kvnevz kinrjB"
        />
        <button
          aria-label="carousel-navigation-button"
          class="sc-kvnevz kinrjB"
        />
        <button
          aria-label="carousel-navigation-button"
          class="sc-kvnevz kinrjB"
        />
        <button
          aria-label="carousel-navigation-button"
          class="sc-kvnevz kinrjB"
        />
        <button
          aria-label="carousel-navigation-button"
          class="sc-kvnevz kinrjB"
        />
        <button
          aria-label="carousel-navigation-button"
          class="sc-kvnevz kinrjB"
        />
        <button
          aria-label="carousel-navigation-button"
          class="sc-kvnevz kinrjB"
        />
        <button
          aria-label="carousel-navigation-button"
          class="sc-kvnevz kinrjB"
        />
      </div>
      <button
        aria-label="next-slide-button"
        class="sc-kcLKEh grrYbN"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-chevron-right"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m9 18 6-6-6-6"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`Carousel component > shows error status when products data is not available 1`] = `
<div>
  <div
    class="sc-dTvVRJ bRYsGP"
  >
    <div
      class="sc-fhHczv sc-ggWZvA ewGzxY MIVzs"
    >
      <p>
        Could not load carousel :(
      </p>
      <p>
        Error: 
        Could not fetch products
      </p>
    </div>
  </div>
</div>
`;

exports[`Carousel component > shows loading status 1`] = `
<div>
  <div
    class="sc-dTvVRJ bRYsGP"
  >
    <div
      class="sc-fhHczv sc-ggWZvA ewGzxY MIVzs"
    >
      Loading products...
    </div>
  </div>
</div>
`;
